### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by Baayeh**
## Building a recipe app

> In this project, @karayamanemre and myself completed the following tasks:

- [x] Created a new rails app and gave it a meaningful name.
- [x] Initialized the project with git.
- [x] Made sure to setup project with PostgreSQL.
- [x] Setup linters.
- [x] Setup Rspec and Capybara testing libraries.
- [x] Installed and setup `devise` gem to:
     - [x] Register a new user.
     - [x] Allow users to login with a combination of email and password.
     - [x] Store hashed passwords in the database.
     - [x] Asked for email confirmation.
     - [x] Allow users to reset their passwords.
- [x] Created a navigation menu that allows users to open all of the pages you created.
- [x] Generated Model, Views and Controller for User.
- [x] Generate Model, Views and Controller for Food.
- [x] Generate Model, Views and Controller for Recipe.
- [x] Generate Model, Views and Controller for RecipeFood.
- [x] Implemented Food List views in order to:
    - [x] display a list of food added by the logged-in user as in the wireframe
    - [x] Display the quantity of a given food.
    - [x] lead to a form that allows users to add new food.
- [x] Implemented Public Recipe list view to:
    - [x] display a list of all public recipes ordered by newest as in the wireframe.
    - [x] lead to recipe details.
- [x] Implemented Recipe list view to:
    - [x] display a list of recipes created by the logged-in user as in the wireframe.
    - [x] lead to recipe details.
    - [x] If the user is the owner of the recipe, should allow the user to delete it.
- [x] Implemented Recipe Details view to:
    - [x] f the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
    - [x] If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- [x] Implemented Shopping List view to:
    - [x] show the list of food that is missing for all recipes of the logged-in user
    - [x] Should `count` the total `food` items and the `total price` of the missing food.
- [x] Made sure there are no N+1 queries happening.

### CODE REVIEW

**Review by youmari — CHANGES_REQUESTED** [review]
@Baayeh I made a mistake here

File: app/views/recipes/_shopping_list.html.erb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<tr>
+  <th scope="row"><%= index + 1 %></th>
+  <td><%= ingredient.food.name %></td>
+  <td><%= ingredient.quantity - ingredient.food.quantity %> <%= ingredient.food.measurement_unit %></td>
+  <td>$<%=ingredient.food.price * (ingredient.quantity - ingredient.food.quantity) %></td>
+</tr>
```
[the remark this answers is not in the document]
Comment (Baayeh) [reply app/views/recipes/_shopping_list.html.erb:6]: I have modified the method to include all missing foods for every recipe

**Review by youmari — APPROVED** [review]
Hi @Baayeh @karayamanemre , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 
## Highlights
✔️ Nice Design
✔️ Clean code 
✔️ PRO PR

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear

File: app/views/recipes/_shopping_list.html.erb (hunk 1 above)
Comment (Baayeh) [reply app/views/recipes/_shopping_list.html.erb:6]: Please find the screenshot of the results.

### Recipes created
image (1)


### First Recipe
image (2)


### Second recipe
image (5)


### Shopping list
image (3)


### Food list
image (4)

**Pull request merged**